IdentityCore/src/oauth2/token/MSIDFamilyRefreshToken.m

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+#import "MSIDFamilyRefreshToken.h"
+
+@implementation MSIDFamilyRefreshToken
+
+- (instancetype)initWithRefreshToken:(MSIDRefreshToken *)refreshToken
+{
+    if (refreshToken && [refreshToken isKindOfClass:[MSIDRefreshToken class]])
```

Comment:
1. Can we do the validation before passing into the init method? And also log the type of the token if type is not expected
2. Please consider to remove duplicate code. e.g. if this is a subclass, maybe the credentialType is only needed?

Reply:
Discussed offline and the change is not duplicated and cannot be avoided based on related changes so far. We do have examples filling in token with response, so this is a similar one to fill in the FRT